### ISSUE TITLE
Reload mode loads infinitely on port 7860 (using the gradio script.py) 

### ISSUE THREAD

**Issue by altryne** [gradio-app/gradio#2102]
### Describe the bug

I run the debug mode, with the hopes of getting the autoreload working, and for some reason, it keeps loading and then times-out. 

I then can access the running server on port 7861 but then every safe, I need to manually go to 7862 or 7863... 

![CleanShot 2022-08-28 at 13 12 38@2x](https://user-images.githubusercontent.com/463317/187090711-a25d619a-d966-4372-b6e2-b1b2046cae9c.jpg)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

I ran `gradio script.py` 
with debug=True in settings 
and no queue

### Screenshot

![CleanShot 2022-08-28 at 13 12 38@2x](https://user-images.githubusercontent.com/463317/187090724-3f36b62f-8ecd-4ee7-a3da-e69fed19d8da.jpg)

### Logs

```shell
no logs for this issue
```

### System Info

```shell
Gradio 3.7.1
```

### Severity

annoying

**Comment by abidlabs** [comment]
Thanks for creating the issue @altryne. Would you be able to attach the `script.py` or another minimal example where this issue is happening?

**Comment by altryne** [comment]
I'll see if I can do a clean repro, but meanwhile, it happens to me on this repo: 
https://github.com/hlky/stable-diffusion-webui
I have exported Gradio interface it it's own module, and when I run 
`gradio webui_playground.py` it shows that it's monitoring the changes and it indeed does, but doesn't load on port 7860 😅

**Comment by verm** [comment]
This happens to me on FreeBSD 13.1 as well with https://github.com/hlky/stable-diffusion-webui 

The OSError returned is:

  ```Port 7860 is in use. If a gradio.Blocks is running on the port, you can close() it or gradio.close_all()```

Interestingly it used to work fine now it does not.  The error is in ```scripts/webui.py```

**Comment by verm** [comment]
I have submitted a an issue with a possible fix for this #2167.

**Comment by freddyaboulton** [comment]
Does this only happen on `stable-diffusion-webui` ?

Doesn't happen with my local scripts. Here is some example output showing it reloaded on the same port, `7861`

```bash
Watching: '/Users/freddy/sources/gradio/gradio', '/Users/freddy/sources/scratch'

Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
WARNING:  StatReload detected changes in 'html_button.py'. Reloading...
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
```

**Comment by abidlabs** [comment]
Hi folks! I closed #2167 because we have not been able to reproduce locally with a simple script. I'll go ahead and close this issue for the same reason. @altryne @verm feel free to reopen if you can share with us a simpler repro, thank you!
